internal/provider/identity_client_server.go

```diff
@@ -179,14 +179,18 @@ func identityClientCreate(d *schema.ResourceData, m interface{}) error {
 	client.Claims = *claims
 
 	// handle secrets
-	secrets := new([]structs.Secret)
+	// Use make instead of new here because the zero value for a slice allocated using new is nil which will cause errors
+	// when calling the API. Zero value when using make is an empty slice, which handles the case of no secrets being supplied
+	secrets := make([]structs.Secret, 0)
 	secretList := d.Get("secret").([]interface{})
 
 	for _, sec := range secretList {
+		log.Printf("! In secrets for loop")
```

Comment:
Does this still need to be here for troubleshooting future issues or was it just for debugging while developing?

Reply:
That should be removed. I must've missed it; I'll update the PR